service/matching/taskListManager.go

```diff
@@ -461,16 +464,21 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 	case result := <-tasksForPoll:
 		if result.syncMatch {
 			c.metricsClient.IncCounter(scope, metrics.PollSuccessWithSyncCounter)
+			c.scope.IncCounter(metrics.PollSuccessWithSyncCounter)
```

Comment:
There is going to be an issue with metrics of this form...

`fetch name:poll_success` will return the total values and the domain specific values too. Its not possible to separate out the aggregate and domain specific series - with the current m3ql language as you cannot say give me series without domain tag. 

I'd suggest removing the legacy aggregate metric and only keep the domain tagged. It would mean changing queries wherever they are used with a `| sum`, but that shouldnt be an issue.

Reply:
yeah. I realized this after I pushed this for code review. I am planning to use DomainAll() for places where I need both domain specific and aggregate

Reply:
Do we really need `domain:all` and dual emission - with the relatively low cardinality we have on domains. 

Counters are easy to aggregate with a sum function.. I'd prefer to leave the use of the all tag just for timers which cannot really be aggregated. Ideally we should use histograms and then we would not need to use the `domain:all` tag anywhere..

Reply:
Good question - I guess, we can always add the domain:all in future if we ever hit the cardinality limit - then I am as well inclined to omit domain:all for now